src/mscorlib/src/System/RtType.cs

```diff
@@ -5466,9 +5466,11 @@ internal Object CreateInstanceDefaultCtor(bool publicOnly, bool skipCheckThis, b
 
                     if (ace.m_ctor != null)
                     {
+#if !FEATURE_CORECLR
                         // Perform security checks if needed
                         if (ace.m_bNeedSecurityCheck)
                             RuntimeMethodHandle.PerformSecurityCheck(instance, ace.m_hCtorMethodHandle, this, (uint)INVOCATION_FLAGS.INVOCATION_FLAGS_CONSTRUCTOR_INVOKE);
```

Comment:
This method is fast path. We should also ifdef out the security checks on the regular path in `RuntimeTypeHandle::CreateInstance`.